shared.c

```diff
@@ -878,6 +897,8 @@ bool appimage_type2_register_in_system(char *path, gboolean verbose)
     return TRUE;
 }
 
+void create_thumbnail(const gchar * appimage_file_path, gboolean verbose);
```

Comment:
That seems like a good idea, we could even narrow it further to `const char const* appimage_file_path`.